### PR TITLE
Ft userauth 158793578

### PR DESCRIPTION
**Pull request by MissGray**
#### What does this PR do?
Has create endpoints with db integrated

#### Description of Task to be completed?
has the following endpoints
POST/api/v2/auth/signup
POST/api/v2/auth/signup

 #### What are the relevant pivotal tracker stories?
#158793578

### CODE REVIEW

File: app/v2/models.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import psycopg2
+from flask import current_app
+from werkzeug.security import generate_password_hash, \
+     check_password_hash
+
+users = []
```
Comment (esirK) [inline app/v2/models.py:6]: Do you need this anymore??

File: app/v2/models.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        self.commit()
+
+    def add(self):
+        self.cur.execute("INSERT INTO users (name, username, email, password) VALUES (%s, %s, %s, %s)", (self.name, self.username, self.email, self.password))
+        self.commit()
+
+    def get_by_id(self, id):
+        self.cur.execute("select * from users where id=%s", (id,))
+        self.commit()
+        user = self.get()
```
Comment (esirK) [inline app/v2/models.py]: are you sure this `user = self.get()` is working??

File: app/v2/models.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    def get_by_username(self, username):
+        self.cur.execute("select * from users where username=%s", (username,))
+        self.commit()
+        user = self.get()
+        if not user:
+            return None
+        return self.make_user(user)
+
+    def make_user(self, user):
+        self.id = user[0]
```
Comment (esirK) [inline app/v2/models.py:70]: what data type is a `user` ??

File: app/v2/views.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+@v2.route('/api/v2/auth/signup', methods = ['POST'])
+def create_user():
+    data = request.get_json()
+    username = data.get('username')
+    name = data.get('name')
+    email = data.get('email')
+    password = data.get('password')
+
+    if not all([username,name,email,password]):
```
Comment (esirK) [inline app/v2/views.py]: this is what the documentation for `all` says
**The all() method returns True when all elements in the given iterable are true. If not, it returns False.**
let `username = " "`. will `all([username])` return true or false??

File: app/v2/views.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    if not validate_email(email):
+        return jsonify({"message" : "Please input a valid email"}), 400
+    return jsonify({"message" : "Account created successfully"}), 201
+
+    
+@v2.route('/api/v2/auth/login', methods = ['POST'])
+def login():
+    if not request.is_json:
+        return jsonify({"msg": "Missing JSON in request"}), 400
+    username = request.json.get('username', None)
```
Comment (esirK) [inline app/v2/views.py]: `get` method returns `None` by default so they is no need for specifying this.

File: config.py [hunk 6]
```diff
@@ -8,7 +8,10 @@ class Config:
     Common configurations
     """ 
     DEBUG = True
-    SECRET_KEY = os.getenv('SECRET_KEY') 
+    SECRET_KEY = "dsfdhghfdhg"
```
Comment (esirK) [inline config.py]: Why did you stop using `environment` variable??

File: app/v2/views.py (hunk 4 above)
Comment (DerKip) [reply app/v2/views.py]: image

Comment (DerKip) [reply app/v2/views.py]: s

Comment (esirK) [reply app/v2/views.py]: from the above image, do you think this validation method would work??

File: app/v2/views.py (hunk 5 above)
Comment (MissGray) [reply app/v2/views.py]: okay..will work on that. thank you

File: app/v2/views.py (hunk 4 above)
Comment (MissGray) [reply app/v2/views.py]: on testing, it works. 
it first checks for blank input
then validates emails.
if all those pass it creates the account
i will check for a better method of validation

**Comment by meshack-mbuvi** [comment]
Check on the reviews made above. They are good.

File: app/v2/models.py (hunk 1 above)
Comment (meshack-mbuvi) [inline app/v2/models.py:6]: Are you sure you need `users=[]`?

File: app/v2/views.py (hunk 4 above)
Comment (MissGray) [reply app/v2/views.py]: I think I have seen what you meant.Thank you. correcting it

File: app/v2/models.py (hunk 1 above)
Comment (MissGray) [reply app/v2/models.py:6]: no. thank you  +1:

**Pull request merged**